apps/dashboard/src/middleware.ts

```diff
@@ -22,12 +22,40 @@ export const config = {
 };
 
 export async function middleware(request: NextRequest) {
+  let cookiesToSet: Record<string, string> | undefined = undefined;
   const { pathname } = request.nextUrl;
   const activeAccount = request.cookies.get(COOKIE_ACTIVE_ACCOUNT)?.value;
   const authCookie = activeAccount
     ? request.cookies.get(COOKIE_PREFIX_TOKEN + getAddress(activeAccount))
     : null;
 
+  // utm collection
+  // NOTE: this is not working for pages with rewrites in next.config.js - (framer pages)
```

Comment:
I wonder if we should just move all the rewrites into the middleware instead of doing it "in front" of middleware?